src/advect1d_vertical.jl

```diff
@@ -0,0 +1,253 @@
+        if (ϕ1[i+1]-ϕ1[i])*(ϕ1[i]-ϕ1[i-1]) > 0
+            δₘϕ[i] = min(abs(δϕ[i]), 2*abs(ϕ1[i]-ϕ1[i-1]), 2*abs(ϕ1[i+1]-ϕ1[i])) * sign(δϕ[i])
+            #δₘϕ[i] = min(abs(δϕ[i-1]), 2*abs(ϕ1[i]-ϕ1[i-1]), 2*abs(ϕ1[i+1]-ϕ1[i])) * sign(δϕ[i-1])
+        else
+            δₘϕ[i] = 0
+        end
+    end
+    
+    for i in 2:nz+5
+        #ϕ₊½[i] = 1/2*(ϕ1[i+1]+ϕ1[i]) - 1/6*(δₘϕ[i+1]-δₘϕ[i])
+        ϕ₊½[i] = ϕ1[i] + 
+                (ΔP[i])/(ΔP[i]+ΔP[i+1])*(ϕ1[i+1]-ϕ1[i]) + 
+                1/(ΔP[i-1]+ΔP[i]+ΔP[i+1]+ΔP[i+2])*(
+                (2*ΔP[i+1]*ΔP[i])/(ΔP[i]+ΔP[i+1])*
+                ((ΔP[i-1]+ΔP[i])/(2*ΔP[i]+ΔP[i+1])-(ΔP[i+2]+ΔP[i+1])/(2*ΔP[i+1]+ΔP[i]))*
+                (ϕ1[i+1]-ϕ1[i]) - 
+                ΔP[i]*(ΔP[i-1]+ΔP[i])/(2*ΔP[i]+ΔP[i+1])*δₘϕ[i+1] +
+                ΔP[i+1]*(ΔP[i+1]+ΔP[i+2])/(ΔP[i]+2*ΔP[i+1])*δₘϕ[i] )
+    end
+    
+    for i in 2:nz+4
+        ϕL[i] = ϕ₊½[i]
+        ϕR[i] = ϕ₊½[i+1]
+    end
+    
+    ## Discontinuity detection
+    for i in 2:nz+5
+        #δ²ϕ[i] = 1/(6*ΔP[i-1]^2)*(ϕ1[i+1]-2*ϕ1[i]+ϕ1[i-1])
+        δ²ϕ[i] = 1/(ΔP[i-1]+ΔP[i]+ΔP[i+1])*((ϕ1[i+1]-ϕ1[i])/(ΔP[i+1]+ΔP[i])-(ϕ1[i]-ϕ1[i-1])/(ΔP[i]+ΔP[i-1]))
+    end
+    
+    for i in 2:nz+5
+        if -δ²ϕ[i+1]*δ²ϕ[i-1]*abs(ϕ1[i+1]-ϕ1[i-1]) - ϵ*min(abs(ϕ1[i+1]),abs(ϕ1[i-1])) > 0
+            #η_tilde[i] = -(δ²ϕ[i+1]-δ²ϕ[i-1]) * (ΔP[i-1]^2) / (ϕ1[i+1]-ϕ1[i-1])
+            η̃[i] = -(δ²ϕ[i+1]-δ²ϕ[i-1])/(ΔP[i+1]+ΔP[i])*(ΔP[i]^3+ΔP[i-1]^3)/(ϕ1[i+1]-ϕ1[i-1])
+        else
+            η̃[i] = 0
+        end
+    end
+    
+    for i in 2:nz+5
+        η[i] = max(0, min(η⁽¹⁾*(η̃[i]-η⁽²⁾), 1))
+    end
+    
+    for i in 2:nz+4
+        ϕLᵈ[i] = ϕ1[i] + 1/2*δₘϕ[i]
+        ϕRᵈ[i] = ϕ1[i+1] + 1/2*δₘϕ[i+1]
+    end
+    
+    for i in 2:nz+4
+        ϕL[i] = ϕL[i]*(1-η[i]) + ϕLᵈ[i]*η[i]
+        ϕR[i] = ϕR[i]*(1-η[i]) + ϕRᵈ[i]*η[i]
+    end
+    
+    ## Monotonicity examination
+    for i in 2:nz+4
+        if (ϕR[i]-ϕ1[i])*(ϕ1[i]-ϕL[i]) <= 0
+            ϕL[i] = ϕ1[i]
+            ϕR[i] = ϕ1[i]
+        elseif (ϕR[i]-ϕL[i])*(ϕ1[i]-1/2*(ϕL[i]+ϕR[i])) >= (ϕR[i]-ϕL[i])^2/6
+            ϕL[i] = 3*ϕ1[i] - 2*ϕR[i]
+        elseif -(ϕR[i]-ϕL[i])^2/6 > (ϕR[i]-ϕL[i])*(ϕ1[i]-1/2*(ϕR[i]+ϕL[i]))
+            ϕR[i] = 3*ϕ1[i] - 2*ϕL[i]
+        end
+    end
+        
+    ## Compute flux
+    for i in 2:nz+4
+        Δϕ[i] = ϕR[i] - ϕL[i]
+    end
+    
+    for i in 2:nz+4
+        ϕ₆[i] = 6*(ϕ1[i] - 1/2*(ϕL[i]+ϕR[i]))
+    end
+    
+    for i in 3:nz+3
+        if U[i-2] >= 0
+            courant = U[i-2]*Δt/ΔP[i+1]
+            FLUX[i-2] = courant * (ϕR[i]-1/2*courant*(Δϕ[i]-(1-2/3*courant)*ϕ₆[i]))
+        else
+            courant = U[i-2]*Δt/ΔP[i]
+            FLUX[i-2] = courant * (ϕL[i+1]-1/2*courant*(Δϕ[i+1]-(1+2/3*courant)*ϕ₆[i+1]))
+        end
+    end
+  
+    for i in 4:nz+3
+        ϕ2[i] = ϕ1[i] + (FLUX[i-3]-FLUX[i-2])
+    end
+    
+    for i in 1:nz
+        s1d_out[i] = ϕ2[i+3] * (ϕ2[i+3] >= 0)
+    end
+    
+    return s1d_out
+end
```

Comment:
Need to add tests!

Reply:
Here is a tutorial about tests: https://www.matecdev.com/posts/julia-testing.html

Reply:
This could also be useful as more of a general overview.